container/kvm/initialisation.go

```diff
@@ -95,24 +89,36 @@ func getRequiredPackages(a string) []string {
 // createPool creates the libvirt storage pool directory. runCmd and chownFunc
 // are here for testing. runCmd so we can check the right shell out calls are
 // made, and chownFunc because we cannot chown unless we are root.
-func createPool(pathfinder func(string) (string, error), runCmd runFunc, chownFunc func(string) error) error {
+func ensurePool(poolInfo *libvirtPool, pathfinder func(string) (string, error), runCmd runFunc, chownFunc func(string) error) error {
 	poolDir, err := guestPath(pathfinder)
 	if err != nil {
 		return errors.Trace(err)
 	}
 
-	if err = definePool(poolDir, runCmd, chownFunc); err != nil {
-		return errors.Trace(err)
-	}
-	if err = buildPool(runCmd); err != nil {
-		return errors.Trace(err)
+        if poolInfo == nil {
+		if err = definePool(poolDir, runCmd, chownFunc); err != nil {
+			return errors.Trace(err)
+		}
+	}else{
+	        logger.Debugf(`pool already created "%#v"`, poolInfo)
```

Comment:
I don't think we need to log the pool struct contents (potentially twice).
Something like:
```go
logger.Debugf(`pool %q already created`, poolInfo.Name)
```
will do.

Reply:
Done